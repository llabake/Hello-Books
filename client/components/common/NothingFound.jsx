import React from 'react';
import PropTypes from 'prop-types';

const NothingFound = ({ text, id }) => {
  return (
    <div
      className="card-panel row center-align"
      id={id}>
      <h4>
        {text}
      </h4>
    </div>
  )
};

NothingFound.propTypes = {
  id: PropTypes.string,
  text: PropTypes.string
};

export default NothingFound;