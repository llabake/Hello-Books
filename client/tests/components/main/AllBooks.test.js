import React from 'react';

import {
  AllBooks,
  mapStateToProps,
  mapDispatchToProps
} from '../../../components/main/AllBooks';

import adminData from '../../mocks/userData';
import userData from '../../mocks/userData';
import {Pagination} from 'react-materialize'
import NothingFound from '../../../components/common/NothingFound'

function setup() {
  const props = {
    dispatch: jest.fn(),
    user: adminData.admin,
    fetchAllBooks: jest.fn(),
    handleLogout: jest.fn(),
    handleShowAllBooks: jest.fn(),
    handleShowReturnRequestList: jest.fn(),
    handleShowBorrowRequestList: jest.fn(),
    errors: {},
    books: [userData.book1, userData.book2],
    bookCount: 2
  };

  const state = {
    redirect: false,
    maxItems: 2,
    showPagination: false,
  };
  const wrapper = shallow(<AllBooks {...props} {...state} />);

  return {
    wrapper,
    props,
    state,
  };
}

describe('All books page', () => {
  test('should render and not crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  })

  test('should call component did mount', () => {
    let spy;
    afterEach(() => {
      spy.mockClear();
    });
    spy = jest.spyOn(AllBooks.prototype, 'componentDidMount')
    const { wrapper, props } = setup();
    expect(wrapper).toBeDefined();
    expect(AllBooks.prototype.componentDidMount).toHaveBeenCalledTimes(1)
  })

  test('should render the book list', () => {
    const { wrapper, props, state } = setup();
    expect(wrapper.find(Pagination).length).toBe(0);
  })

  test('should render nothing found component', () => {
    const props = {
      dispatch: jest.fn(),
      user: adminData.admin,
      errors: {},
      books: [],
      bookCount: 0,
      fetchAllBooks: jest.fn(() => Promise.resolve(1)),
    }
    const wrapper = shallow(<AllBooks {...props} />);

    expect(wrapper.find(NothingFound).length).toBe(1)

  })

  test('should call handle selected page', () => {
    const { wrapper, props, state } = setup();
    const page = 2
    // wrapper.setState({
    //   activePage: page,
    // });
    wrapper.instance().handleSelectedPage(page)
    // expect(wrapper.instance().state.activePage).toBe(2);
  });

  test('should call componentWillReceiveProps', () => {
    let spy;
    afterEach(() => {
      spy.mockClear();
    });

    const props = {
      dispatch: jest.fn(),
      user: adminData.admin,
      errors: {},
      books: [],
      bookCount: 0,
      fetchAllBooks: jest.fn(() => Promise.resolve(1)),
    }

    const nextProps = {
      dispatch: jest.fn(),
      user: adminData.admin,
      errors: {},
      books: [],
      bookCount: 0,
      fetchAllBooks: jest.fn(() => Promise.resolve(1)),
    }
    spy = jest.spyOn(AllBooks.prototype, 'componentWillReceiveProps');
    const wrapper = shallow(<AllBooks {...props} />)
    wrapper.setProps(nextProps);
    expect(AllBooks.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1)
    expect(spy).toHaveBeenCalled();
  })


  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('fetchAllBooks');

    const { fetchAllBooks } = mapDispatchToProps(dispatch);
    fetchAllBooks();
    expect(dispatch).toHaveBeenCalled();
  });

}) 

