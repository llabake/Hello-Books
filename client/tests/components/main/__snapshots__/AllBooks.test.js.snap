// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All books page should render and not crash 1`] = `
<div>
  <div
    className="container"
  >
    <div>
      <div
        className="row "
      >
        <div
          className="section"
        >
          <div
            className="row"
            id="allbooks-list"
          >
            <div
              key="2"
            >
              <BookCard
                book={
                  Object {
                    "aboutAuthor": "alrefi is a god writer ten ",
                    "author": "Aaidh ibn Abdullah al-Qarni",
                    "borrowCount": 2,
                    "createdAt": "2018-03-12T14:51:55.457Z",
                    "description": "dont be sad life is good ",
                    "downVotes": 0,
                    "id": 2,
                    "image": "https://res.cloudinary.com/sardaunan/image/upload/v1520866315/hbg1rryhpzabtnh2akzx.webpk",
                    "isbn": "298848",
                    "publishedYear": "2000",
                    "quantity": 4,
                    "title": "dont be sad ",
                    "upVotes": 1,
                    "updatedAt": "2018-03-14T17:21:14.663Z",
                  }
                }
                id={2}
              />
            </div>
            <div
              key="3"
            >
              <BookCard
                book={
                  Object {
                    "aboutAuthor": "KEITH FERRAZZI is founder and CEO of the training and consulting company Ferrazzi Greenlight and a contributor to Inc., the Wall Street Journal, and Harvard Business Review. Earlier in his career, he was CMO of Deloitte Consulting and at Starwood Hotels and Resorts, and CEO of YaYa Media. He lives in Los Angeles.",
                    "author": "Keith Ferrazi",
                    "borrowCount": 2,
                    "createdAt": "2018-03-12T15:27:19.859Z",
                    "description": "\\"Everyone in business knows relationships and having a network of contacts is important. Finally we have a real-world guide to how to create your own high-powered network tailored to your career goals and personal style.\\"",
                    "downVotes": 1,
                    "id": 3,
                    "image": "https://res.cloudinary.com/sardaunan/image/upload/v1520868439/rrtw8mjav6q9ntjnysvgk.jpg",
                    "isbn": "385346654",
                    "publishedYear": "2014",
                    "quantity": 7,
                    "title": "Never eat alone ",
                    "upVotes": 0,
                    "updatedAt": "2018-03-14T17:21:13.503Z",
                  }
                }
                id={3}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
